peer/peerstorage.go

```diff
@@ -0,0 +1,128 @@
+package peer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+var (
+	// peerStorage is the key that maps the peer's storage to its
+	// public key in the Db.
+	peerStorage = []byte("peer-storage")
+
+	// ErrUninitializedDB is called when the DB is not initialized yet
+	// but a caller calls the `RetrievePeerData`.
+	ErrUninitializedDB = errors.New("uninitialized Kvdb peer data " +
+		"store")
+)
+
+// KvdbPeerStorage is the kvdb implementation of the DataStoreForPeer interface.
+type KvdbPeerStorage struct {
+	// DB is th kvdb backend for the storage layer.
+	DB kvdb.Backend
+}
+
+// KvdbPeerStorageConfig is the configuration detail for the kvdb storage layer.
+type KvdbPeerStorageConfig struct {
+	// DB is th kvdb backend for the storage layer.
+	DB kvdb.Backend
+}
+
+// NewKvdbPeerStorage returns an instance of the kvdb storage layer.
+func NewKvdbPeerStorage(c *KvdbPeerStorageConfig) (*KvdbPeerStorage, error) {
+	return &KvdbPeerStorage{
+		DB: c.DB,
+	}, nil
+}
+
+// Start initializes the storage.
+func (k *KvdbPeerStorage) Start() error {
+	err := kvdb.Update(k.DB, func(tx kvdb.RwTx) error {
+		_, err := tx.CreateTopLevelBucket(peerStorage)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}, func() {})
+
+	if err != nil {
+		err2 := k.DB.Close()
+		if err != nil {
+			return fmt.Errorf(
+				"error initializing peerStorage DB: %v"+
+					" and error closing peerStorage DB: "+
+					"%v", err, err2)
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+// Stop close the database connection for the storage.
+func (k *KvdbPeerStorage) Stop() error {
+	return k.DB.Close()
+}
```

Comment:
Same here. I just looked into this too, we pretty much never call this method (`Close()`) outside of tests. The cleanup is handled by

https://github.com/lightningnetwork/lnd/blob/d5138c10391b5ba4b9f63dda7e3416c4da0a8de5/lncfg/db.go#L227